tendrl/commons/utils/central_store/utils.py

```diff
@@ -107,7 +107,8 @@ def watch(obj, key):
                     attr = key.rstrip("/").split("/")[-1]
                     obj.on_change(attr, prev_val,
                                   cur_val)
-        except etcd.EtcdKeyNotFound:
+        except Exception:
+            # When watch crash then clear key from the watchers
```

Comment:
Sometimes watcher is crashing when a key is deleted by TTL (but not for all the time, need to figure out why),  we have to remove the key to create new watcher when save function called for that particular object.